### PR TITLE
cpu/cortex-m: add all-debug target

### PR DESCRIPTION
**Pull request by DipSwitch**
Request to include the Makefile target `all-debug` for the cortex-m serie CPU's.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@LudwigKnuepfer no job for you? I'd like to assign you ;-) !

**Comment by LudwigKnuepfer** [comment]
I don't quite understand - I thought debugging symbols are always included in the hexfile without the need for `-g` ..?

**Comment by OlegHahm** [comment]
I also thought so.

**Comment by DipSwitch** [comment]
Oops, this is true. But I would like to see the `-Os` disabled. :)

**Comment by DipSwitch** [comment]
And it would be nice to have the target itself existing aswell.

**Comment by kaspar030** [comment]
@DipSwitch ```CFLAGS_OPT=-O0 make all```?

**Comment by OlegHahm** [comment]
> And it would be nice to have the target itself existing aswell.

For consistency with native, this might make sense.

**Comment by DipSwitch** [comment]
This way you don't have to change the default build target in your IDE if you switch between native and a board =)

File: cpu/cortexm_common/Makefile.include [hunk 1]
```diff
@@ -1,2 +1,6 @@
 # include module specific includes
 export INCLUDES += -I$(RIOTCPU)/cortexm_common/include
+
+all-debug: export CFLAGS += -g
```
Comment (LudwigKnuepfer) [inline cpu/cortexm_common/Makefile.include]: So, will you remove this superfluous line? Then we can merge.

Comment (LudwigKnuepfer) [inline cpu/cortexm_common/Makefile.include]: On second thought: it probably makes more sense to move the all-debug dependency declaration to some global location and purge it here and from the respective location in native.

**Comment by kaspar030** [comment]
> @DipSwitch CFLAGS_OPT=-O0 make all?

Again, why not override on command line?
```CFLAGS_OPT=-O0 -g make all``` should be honoured by all targets and do exactly what you want. Is there an advantage having it as a specific make target?

**Comment by DipSwitch** [comment]
@kaspar 30 Because there is no tab completion ^_^

Sorry, i just think it would be nice for all boards to have the same make
targets. Just like programming api's :) we could also remove the all-debug
from native :)

**Comment by DipSwitch** [comment]
Removed line, but maybe it's better to move it to a general location so also the MSP430 can provide from this.

**Comment by LudwigKnuepfer** [comment]
> Removed line, but maybe it's better to move it to a general location so also the MSP430 can provide from this.

I agree (and said so ;))

**Comment by DipSwitch** [comment]
Sorry, but where to add? ;) home in a minute.

**Comment by LudwigKnuepfer** [comment]
How about `Makefile.include`?

**Comment by DipSwitch** [comment]
Followup PR made.

**Pull request closed without merging**